### DISCUSSION TITLE
Testing understanding - reset after trigger off

### DISCUSSION THREAD

**Discussion by bowana** [MiguelRipoll23/homebridge-securitysystem#524] [Q&A]
Hi, thanks for the effort that has gone into this project - I look forward to a potential matter version in the future.

I just wanted to check my understanding of how the alarm is meant to work after it has been tripped.  

In my scenario, the alarm is armed in “home” mode.  A door sensor turns the Trip Home Switch “on” and after a delay the alarm is triggered.  Great!
When the door is closed, an automation turns the Trip Home Switch “off”.  When this occurs, the alarm immediately goes into “off” mode and (at the moment) requires manual arming.  Is there a way to have the alarm return to “home” mode in this scenario? Or perhaps I’m not quite understanding the logic around the alarm reset? (Which only appears to occur after being triggered for a period of time).

Appreciate the guidance.  

Tim

**Comment by MiguelRipoll23** [comment]
Thanks for the words.

As I understood, you want to set the mode Home if door closes when the security system is triggered. I’d convert the home automation to a Shortcut that uses the action “Get the state of My Home” to get the current security system mode and check using an “If” condition if it’s currently triggered.

Finally, the action to control the home and set the mode back to Home.
